geomstats/datasets/sample_sdp_2d.py

```diff
@@ -0,0 +1,131 @@
+"""Helper data classes for the MDM illustration example on SPD matrices."""
+
+import geomstats.backend as gs
+from geomstats.geometry.spd_matrices import SPDMatrices
+from geomstats.geometry.special_orthogonal import SpecialOrthogonal
+
+
+class DatasetSPD_2D():
+    """
+    Sample 2D SPD dataset.
+
+    Data is of shape [n_samples * n_classes, n_features];
+    Labels are of shape [n_samples * n_classes, n_classes].
+    """
+
+    def __init__(self, n_samples=100, n_features=2, n_classes=3):
+        self.n_samples = n_samples
+        self.n_features = n_features
+        self.n_classes = n_classes
+        self.data_helper = DataHelper()
+
+    def generate_sample_dataset(self):
+        """
+        Generate the dataset.
+
+        :return:
+        X: array-like, shape = [n_samples * n_classes, n_features]: data
+        Y: array-like, shape = [n_samples * n_classes, n_classes]: labels
+        """
+        X, Y = self.setup_data()
+        X, Y = self.data_helper.shuffle(X, Y)
+        return X, Y
+
+    def setup_data(self):
+        """
+        Generate the un-shuffled dataset.
+
+        :return:
+        X: array-like, shape = [n_samples * n_classes, n_features]: data
+        Y: array-like, shape = [n_samples * n_classes, n_classes]: labels
+        """
+        mean_covariance_eigenvalues = gs.random.uniform(
+            0.1, 5., (self.n_classes, self.n_features))
+        base_rotations = SpecialOrthogonal(n=self.n_features).random_gaussian(
+            gs.eye(self.n_features), 1, n_samples=self.n_classes)
+        # var_eigenvalues = gs.random.uniform(
+        #     .04, .06, (self.n_classes, self.n_features))
+        var_rotations = gs.random.uniform(
+            .5, .75, (self.n_classes))
+
+        # data
```

Comment:
While comments are fine in the examples, we tend to avoid them in the core codebase. Comments are difficult to maintain with several developers, it often happens that a contributor will change some part of the code and forget to change the comment that becomes deprecated.

If it feels like a portion of the code needs a comment, maybe it means that the variable names should be more explicit, or that this portion of code should be factorized into an auxiliary function, with a docstring and an explicit name.

Reply:
Clarified notations (using scikit-learn formatting).